src/shogun/mathematics/linalg/GPUBackend.cpp

```diff
@@ -37,6 +37,7 @@
 #ifdef HAVE_VIENNACL
 #include <viennacl/vector.hpp>
 #include <viennacl/linalg/inner_prod.hpp>
+#include <viennacl/linalg/sum.hpp>
```

Comment:
why close here the `#ifdef HAVE_VIENNACL` ?
i mean if you don't have viennacl this whole implementation is just foobar.
do it like
```
#ifdef HAVE_VIENNACL

... the whole GPUBackend implementation...
#endif
```